src/Microsoft.Extensions.Logging.Abstractions/LoggerMessage.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging.Internal;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Logging
+{
+    /// <summary>
+    /// Creates delegates which can be later cached to log messages in a performant way.
+    /// </summary>
+    public static class LoggerMessage
+    {
+        /// <summary>
+        /// Creates a delegate which can be used for creating a log scope.
+        /// </summary>
+        /// <param name="scope">A delegate which creates a log scope</param>
+        /// <param name="formatString">The named format string</param>
+        public static void DefineScope(out Func<ILogger, IDisposable> scope, string formatString)
+        {
+            var formatter = new LogValuesFormatter(formatString);
+
+            scope = logger => logger.BeginScopeImpl(new LogValues(formatter));
```

Comment:
Should we close over `new LogValues` as well? In the case of just a format string we can remove that allocation